examples/chef/create_docker.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import tarfile
+import shutil
+import docker
+
+client = docker.from_env()
+
+_CREATE_DOCKER_SCRIPT_PATH = os.path.dirname(__file__)
+_SUPPORTED_PLATFORM = 'linux_x86'
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-c', '--commit_sha', required=True,
+                    action='store',
+                    dest='commit_sha')
+parser.add_argument('-s', '--short_sha', required=True,
+                    action='store',
+                    dest='short_sha')
+parser.add_argument('-r', '--revision_id', required=True,
+                    action='store',
+                    dest='revision_id')
+parser.add_argument('-b', '--build_id', required=True,
+                    action='store',
+                    dest='build_id')
+parser.add_argument('-i', '--image_name', required=True,
+                    action='store',
+                    dest='image_name')
+parser.add_argument('-t', '--tar_path', required=True,
+                    action='store',
+                    dest='tar_path')
+
+args = parser.parse_args()
+
+out_directory = f'{_CREATE_DOCKER_SCRIPT_PATH}/out'
+
+for device_file_name in os.listdir(args.tar_path):
+    # Clean up the out directory before extracting device files
+    shutil.rmtree(out_directory)
```

Comment:
maybe move th rm/mkdir after the supported platform split, so that we do not do any work if the tar file does not match expected format